github/src/main/java/io/syndesis/github/GitHubServiceImpl.java

```diff
@@ -79,7 +83,24 @@ public String getCloneURL(String repoName) throws IOException {
 
     @Override
     public User getApiUser() throws IOException {
-        return userService.getUser();
+        final User user = userService.getUser();
+        // if the user did not elect to publicly display his e-mail address, e-mail will be null
+        // https://developer.github.com/v3/users/#get-a-single-user
+        // let's put a dummy e-mail address then, as it is needed for the commit
+        if (user.getEmail() == null) {
+            // users before 2017-07-18 have their no-reply e-mail addresses in the form
+            // username@users.noreply.github.com, and after that date
+            // id+username@users.noreply.github.com
+            // https://help.github.com/articles/about-commit-email-addresses/
```

Comment:
wow, cool finding. And also many thanks for commenting the details (although the urls might get stale sometimes, but really appreciate it)